themes/bootstrap5/templates/Recommend/SideFacets/range-slider.phtml

```diff
@@ -1,9 +1,9 @@
 <?php
   // Sanitize inputs:
   $sanitize = match($this->facet['type']) {
-    'date' => fn ($val) => preg_replace('/[^\d]*/', '', $val), // digits only for year-only dates
+    'date' => fn ($val) => preg_replace('/^.*?(-?\d+).*?^/', '$1', $val), // digits/sign only for year-only dates
```

Comment:
Not sure I understand the `.*?` construct here. Since `.*` means "0 or more characters", what does the `?` add to this?

Reply:
Also, why are there carets on both ends of this?

Reply:
The question mark makes it non-eager. Since a single digit satisfies the `(-?\d+)` part, `.*` would match everything up to the last digit. I.e. with `2025` as the input, `.*` would match `202` and only leave `5` for the capturing group.
The caret at the end was a mistake.